api/mesh/v1alpha1/kds.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package kuma.mesh.v1alpha1;
+
+option go_package = "v1alpha1";
+
+import "envoy/api/v2/discovery.proto";
+import "google/protobuf/any.proto";
+
+service KumaDiscoveryService {
+
+  rpc DeltaKumaResources(stream envoy.api.v2.DeltaDiscoveryRequest)
+      returns (stream envoy.api.v2.DeltaDiscoveryResponse);
+
+  rpc StreamKumaResources(stream envoy.api.v2.DiscoveryRequest)
+      returns (stream envoy.api.v2.DiscoveryResponse);
+
+  rpc FetchKumaResources(envoy.api.v2.DiscoveryRequest)
+      returns (envoy.api.v2.DiscoveryResponse);
+}
+
+message KumaResource {
+  message Meta {
+    string name = 1;
+    string mesh = 2;
```

Comment:
we need creation time and modification time here and maybe a version depending how sync is implemented..